src/core/linux/SDL_dbus.c

```diff
@@ -359,8 +377,68 @@ SDL_DBus_ScreensaverTickle(void)
 SDL_bool
 SDL_DBus_ScreensaverInhibit(SDL_bool inhibit)
 {
-    if ( (inhibit && (screensaver_cookie != 0)) || (!inhibit && (screensaver_cookie == 0)) ) {
+    const char *default_inhibit_reason = "Playing a game";
+
+    if ( (inhibit && (screensaver_cookie != 0 || inhibit_handle != NULL))
+        || (!inhibit && (screensaver_cookie == 0 && inhibit_handle == NULL)) ) {
         return SDL_TRUE;
+    }
+
+    /* Check if we are running inside a Flatpak or Snap container.
+     * For Snap, we check multiple variables because they might be set for unrelated
+     * reasons. This is the same thing WebKitGTK does. */
+    if (access("/.flatpak-info", F_OK) == 0
+        || (SDL_getenv("SNAP") != NULL && SDL_getenv("SNAP_NAME") != NULL && SDL_getenv("SNAP_REVISION") != NULL)) {
+
+        const char *node = "org.freedesktop.portal.Desktop";
+        const char *path = "/org/freedesktop/portal/desktop";
+        const char *interface = "org.freedesktop.portal.Inhibit";
+        const char *window = ""; /* As a future improvement we could gather the X11 XID or Wayland surface identifier */
+        static unsigned int INHIBIT_IDLE = 8; /* Taken from the portal API reference */
+        DBusMessageIter iterInit, iterDict, iterEntry, iterValue;
+
+        if (inhibit) {
+            DBusMessage *msg;
+            SDL_bool retval = SDL_FALSE;
+            const char *key = "reason";
+            char *reply = NULL;
```

Comment:
Perhaps better as

```suggestion
            const char *reply = NULL;
```

as a hint that it's a "borrowed" pointer pointing into the message buffer, so SDL would be wrong to free or modify the string it points to.